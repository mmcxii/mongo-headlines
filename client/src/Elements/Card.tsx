import styled from 'styled-components';

import { rounded, spacing, elevation, transition, sarawackWhitePaper, blueBell } from 'Utilities';

export const Card = styled.section`
    border-radius: ${rounded};
    background: ${sarawackWhitePaper};
    ${elevation[3]};
    ${transition({ prop: 'box-shadow' })};

    &:hover {
        ${elevation[4]};
    }
`;

export const CardHeader = styled.h3`
    padding: ${spacing.sm} 0;
    margin: 0 ${spacing.md};
    border-bottom: 2px solid ${blueBell};
    color: ${blueBell};
`;

export const CardBody = styled.div`
    font-size: 1.05rem;
    padding: ${spacing.md};
    display: flex;
    flex-direction: column;
`;
