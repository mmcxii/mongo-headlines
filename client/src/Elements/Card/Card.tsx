import styled from 'styled-components';

import { rounded, spacing, elevation, transition, cityLights, blueBell } from 'Utilities';

export const Card = styled.section`
    border-radius: ${rounded};
    background: ${cityLights};
    ${elevation[3]};
    ${transition({ prop: 'box-shadow' })};

    &:hover {
        ${elevation[4]};
    }
`;

export const CardHeader = styled.h3`
    padding: ${spacing.sm} 0;
    margin: 0 ${spacing.md};
    border-bottom: 2px solid ${blueBell};
    color: ${blueBell};

    > a {
        color: inherit;
        text-decoration: none;

        &:hover {
            text-decoration: underline;
        }
    }
`;

export const CardBody = styled.div`
    font-size: 1.05rem;
    padding: ${spacing.md};
    display: flex;
    flex-direction: column;
`;
