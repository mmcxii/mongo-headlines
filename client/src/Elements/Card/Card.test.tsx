import React from 'react';
import { render } from '@testing-library/react';

import { Card, CardHeader, CardBody } from './Card';

describe('<Card /> tests', () => {
    it('should render a card with a header and body', () => {
        //* Act
        const { container } = render(
            <Card>
                <CardHeader>Test Card</CardHeader>
                <CardBody>
                    <p>
                        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Ipsam dolores esse harum
                        voluptates laborum autem necessitatibus amet eveniet quo incidunt, ab inventore quidem
                        at! Delectus sed molestias accusantium libero eos.
                    </p>
                </CardBody>
            </Card>
        );

        //* Assert
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render a card as an article with an h2 header and a body as a p', () => {
        //* Act
        const { container } = render(
            <Card as='article'>
                <CardHeader as='h2'>Test Card Two</CardHeader>
                <CardBody as='p'>
                    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Officiis modi velit commodi!
                    Ullam est odit in beatae ratione qui ex exercitationem ipsam explicabo, quidem quas,
                    commodi magni. Ratione, tempora. Aperiam.
                </CardBody>
            </Card>
        );

        //* Assert
        expect(container.firstChild).toMatchSnapshot();
    });
});
