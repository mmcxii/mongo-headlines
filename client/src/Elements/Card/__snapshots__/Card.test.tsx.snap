// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> tests should render a card as an article with an h2 header and a body as a p 1`] = `
<article
  class="sc-bdVaJa cmRAvc"
>
  <h2
    class="sc-bwzfXH dxWcPK"
  >
    Test Card Two
  </h2>
  <p
    class="sc-htpNat hmHgGj"
  >
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Officiis modi velit commodi! Ullam est odit in beatae ratione qui ex exercitationem ipsam explicabo, quidem quas, commodi magni. Ratione, tempora. Aperiam.
  </p>
</article>
`;

exports[`<Card /> tests should render a card with a header and body 1`] = `
<section
  class="sc-bdVaJa cmRAvc"
>
  <h3
    class="sc-bwzfXH dxWcPK"
  >
    Test Card
  </h3>
  <div
    class="sc-htpNat hmHgGj"
  >
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Ipsam dolores esse harum voluptates laborum autem necessitatibus amet eveniet quo incidunt, ab inventore quidem at! Delectus sed molestias accusantium libero eos.
    </p>
  </div>
</section>
`;
